pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -383,35 +393,29 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			}
 
 			buildRun.Status.LatestTaskRunRef = &lastTaskRun.Name
-			buildRun.Status.StartTime = lastTaskRun.Status.StartTime
+
+			if buildRun.Status.StartTime == nil {
```

Comment:
In Tekton's TaskRun, `Status.StartTime` is defined as `*metav1.Time` and therefore can be nil. We might only be hitting this case here if a TaskRun fails right away and is never started. If my understanding of the `IsZero` function is correct, I think we could do:

```go
if buildRun.Status.StartTime == nil && !lastTaskRun.Status.StartTime.IsZero() {
```

As I do not think that Tekton will ever set a zero timestamp, this will probably work the same way:

```go
if buildRun.Status.StartTime == nil && lastTaskRun.Status.StartTime != nil {
```

Reply:
Good point. I will add the second check.